### PR TITLE
[API] Support partitioning in list runs endpoint

### PR DESCRIPTION
**Pull request by Hedingber**
Also did some name alignments on the feature store's partition params

Part of the implementation of https://jira.iguazeng.com/browse/ML-1397

### CODE REVIEW

File: mlrun/api/schemas/constants.py [hunk 1]
```diff
@@ -91,10 +91,36 @@ def to_partition_by_db_field(self, db_cls):
...
+    def to_partition_by_db_field(self, db_cls):
+        if self.value == RunPartitionByField.name:
+            return db_cls.name
+        else:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Unknown group by field: {self.value}"
+            )
+
+
 # For now, we only support sorting by updated field
```
Comment (theSaarco) [inline mlrun/api/schemas/constants.py:106]: Please remove this comment.

File: mlrun/db/httpdb.py [hunk 2]
```diff
@@ -1509,7 +1531,9 @@ def list_entities(
         return resp.json()["entities"]
 
     @staticmethod
-    def _generate_partition_by_params(partition_by, rows_per_partition, sort_by, order):
+    def _generate_partition_by_params(
+        partition_by_cls, partition_by, rows_per_partition, sort_by, order
```
Comment (theSaarco) [inline mlrun/db/httpdb.py:1535]: You added this parameter but are not using it. Looks like you forgot to implement this method.

File: mlrun/db/httpdb.py [hunk 3]
```diff
@@ -1509,7 +1531,9 @@ def list_entities(
...
+    ):
         if isinstance(partition_by, schemas.FeatureStorePartitionByField):
```
Comment (theSaarco) [inline mlrun/db/httpdb.py:1537]:
```suggestion
        if isinstance(partition_by, partition_by_cls):
```
Was this what you meant to do? Maybe also assert if this is wrong?

**Review by theSaarco — CHANGES_REQUESTED** [review]
Looks good except for the `httpdb` implementation where you forgot part of the implementation.
Might be a good idea to also add a test going through `httpdb` - this would have caught this issue.

File: mlrun/db/httpdb.py (hunk 3 above)
Comment (Hedingber) [reply mlrun/db/httpdb.py:1537]: Yes, don't think it's needed

**Review by theSaarco — APPROVED** [review]
LGTM

**Pull request merged**